### PR TITLE
Refactor cell params hierarchy

### PR DESCRIPTION
**Pull request by rodriguez-facundo**
refactor cellParams navigation according to mockup and feedback from sprint.

I also added some refactoring to make the code more readable.

travis might not pass because development has been updated with appbar PR. I will work on travis on Tuesday

### CODE REVIEW

**Comment by rodriguez-facundo** [comment]
I think it is ready now. The gear button was inside sections, so I removed and relaunched Travis

File: components/definition/cellRules/NetPyNECellRules.js [hunk 1]
```diff
@@ -361,179 +418,207 @@ export default class NetPyNECellRules extends React.Component {
...
+
+            <NavigationChevronRight style={styles.rightArrow}/>
+
+            <FloatingActionButton
+              zDepth={1}
+              id="newCellRuleButton"
+              style={styles.cellRule}
+              className={"actionButton smallActionButton"}
+              secondary={ page != 'main' }
+              data-tooltip={
```
Comment (adrianq) [inline components/definition/cellRules/NetPyNECellRules.js]: Can we change this to a classic if/else? Probably you can even extract it to a method and maybe reuse in several places?

File: components/definition/cellRules/NetPyNECellRules.js [hunk 2]
```diff
@@ -361,179 +418,207 @@ export default class NetPyNECellRules extends React.Component {
...
-          </div>
-          <div className={"details"}>
-            {selectedSection}
-            {dialogPop}
-          </div>
-        </CardText>
-      );
+              onClick={() => this.handleHierarchyClick('main')}
+            >
+              { page != 'main'
```
Comment (adrianq) [inline components/definition/cellRules/NetPyNECellRules.js]: Can we change this to a classic if/else? Probably you can even extract it to a method and maybe reuse in several places?

File: components/definition/cellRules/NetPyNECellRules.js [hunk 3]
```diff
@@ -361,179 +418,207 @@ export default class NetPyNECellRules extends React.Component {
...
+            <RaisedButton
+              id="newSectionButton"
+              style={styles.sections.container}
+              disabledBackgroundColor="grey"
+              buttonStyle={styles.sections.icon}
+              primary={ page != 'mechanisms' }
+              secondary={ page == 'mechanisms' }
+              disabled={ selectedCellRule == undefined }
+              onClick={ () => this.handleHierarchyClick('sections') }
+              data-tooltip={ page == 'mechanisms'
```
Comment (adrianq) [inline components/definition/cellRules/NetPyNECellRules.js]: Can we change this to a classic if/else? Probably you can even extract it to a method and maybe reuse in several places? Too difficult to understand what is going on

File: components/definition/cellRules/NetPyNECellRules.js [hunk 4]
```diff
@@ -361,179 +418,207 @@ export default class NetPyNECellRules extends React.Component {
...
+                  : !!selectedCellRule
+                    ?  !!model && !!model[selectedCellRule] && Object.keys(model[selectedCellRule]['secs']).length > 0
+                      ? "Explore sections"
+                      : "Create first section"
+                    : "No rule selected" 
+              }
+            >
+              <p style={{color: 'white', height: '100%'}}>
+                {page == "mechanisms" 
+                  ? selectedSection != undefined
```
Comment (adrianq) [inline components/definition/cellRules/NetPyNECellRules.js]: Can we change this to a classic if/else? Probably you can even extract it to a method and maybe reuse in several places?

File: components/definition/cellRules/NetPyNECellRules.js [hunk 5]
```diff
@@ -361,179 +418,207 @@ export default class NetPyNECellRules extends React.Component {
...
-          <div className={"details"}>
-            {selectedMechanism}
-            {dialogPop}
-          </div>
-        </CardText>
-      );
-
-    }
+          <div style={{ clear: "both" }}/>
+          { page == "main" ? cellRulesContainer : page == "sections" ? sectionsContainer : mechanismsContainer }
```
Comment (adrianq) [inline components/definition/cellRules/NetPyNECellRules.js]: Can we change this to a classic if/else? Probably you can even extract it to a method and maybe reuse in several places?

File: components/definition/cellRules/sections/NetPyNESectionThumbnail.js [hunk 6]
```diff
@@ -11,59 +16,50 @@ export default class NetPyNESectionThumbnail extends React.Component {
...
-          {(this.state.isHovered && this.props.selected) ? <FontIcon className="fa fa-trash-o" color="white" hoverColor="white"/> : this.props.name}
+        style={ styles.btn }
+        buttonStyle={ styles.btn }
+        onMouseEnter={ () => this.setState({isHovered: true}) }
+        onMouseLeave={ () => this.setState({isHovered: false}) }
+        data-tooltip={isHovered && name.length > 14 ? name : undefined}
+        className={"rectangularActionButton " + (selected ? "selectedRectangularActionButton " : "")} 
+        onClick={() => this.handleClick()}
+      >
+        {(isHovered && selected)
```
Comment (adrianq) [inline components/definition/cellRules/sections/NetPyNESectionThumbnail.js]: Can we change this to a classic if/else? Probably you can even extract it to a method and maybe reuse in several places?

File: components/definition/cellRules/sections/mechanisms/NetPyNEMechanismThumbnail.js [hunk 7]
```diff
@@ -50,26 +48,33 @@ export default class NetPyNEMechanismThumbnail extends React.Component {
...
+          data-tooltip={isHovered && name.length > 10 ? name : undefined}
+          className={"gearThumbButton " + (selected ? "selectedGearButton" : "")}
+          onClick={ (e, v) => this.handleClick(v) }
+        >
           <div>
-           {(this.state.isHovered && this.props.selected) ? 
-            <FontIcon className="fa fa-trash-o" color="white" hoverColor="white"  style={{zIndex:10,marginTop:37,marginLeft:40,position:"absolute",backgroundColor:"#f23d7a"}}/> :
-            <span className={"gearThumbLabel"}> {this.props.name}</span>}
-           <FontIcon color={changeColor} hoverColor={hoverColor} className="gpt-fullgear"/>
+            {(isHovered && selected)
```
Comment (adrianq) [inline components/definition/cellRules/sections/mechanisms/NetPyNEMechanismThumbnail.js]: Can we change this to a classic if/else? Probably you can even extract it to a method and maybe reuse in several places?

File: components/definition/cellRules/sections/mechanisms/NetPyNENewMechanism.js [hunk 8]
```diff
@@ -14,56 +32,70 @@ export default class NetPyNENewMechanism extends React.Component {
...
-    Utils
-      .evalPythonMessage("netpyne_geppetto.getAvailableMechs", [])
-      .then((response) => {
-        var menuItems = []
-        response.forEach((item) => 
-          menuItems.push(<MenuItem id={item} key={item} value={item} primaryText={item}/>)
-        )
-        this.setState({mechanisms: menuItems})
-      })
+  async componentDidMount() {
```
Comment (adrianq) [inline components/definition/cellRules/sections/mechanisms/NetPyNENewMechanism.js]: No need for this async/await. Just do a regular `evalPythonMessage.then`

File: components/definition/cellRules/sections/mechanisms/NetPyNENewMechanism.js [hunk 9]
```diff
@@ -14,56 +32,70 @@ export default class NetPyNENewMechanism extends React.Component {
...
+        onClick={ e => this.handleButtonClick(e.currentTarget) }
+      >
+        <FontIcon 
+          style={{position: 'absolute'}}
+          className="gpt-fullgear"
+          color={changeColor} 
+          hoverColor={hoverColor} 
+        />
+        
+        {disabled
```
Comment (adrianq) [inline components/definition/cellRules/sections/mechanisms/NetPyNENewMechanism.js]: Can we change this to a classic if/else? Probably you can even extract it to a method and maybe reuse in several places?

File: components/general/CustomIcons.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import SvgIcon from 'material-ui/SvgIcon';
+
+const HomeIcon = (props) => (
```
Comment (adrianq) [inline components/general/CustomIcons.js]: Remove this CustomIcons class. I know this icon is being used in two different files but I don't think it is a good idea to start a CustomIcons for now

Comment (adrianq) [reply components/general/CustomIcons.js]: Btw why don't we use a regular material ui icon? See https://material.io/tools/icons/?icon=home&style=baseline

File: components/general/NetPyNEThumbnail.js [hunk 11]
```diff
@@ -46,21 +32,25 @@ export default class NetPyNEThumbnail extends React.Component {
...
-            {(this.state.isHovered && this.props.selected) ? "" : this.props.name}
+            id={name}
+            onMouseEnter={() => this.setState({isHovered: true})}
+            onMouseLeave={() => this.setState({isHovered: false})}
+            data-tooltip={isHovered && name.length > 14 ? name : undefined}
+            iconClassName={(this.state.isHovered && selected) ? "fa fa-trash-o" : ""} 
+            className={"actionButton " + (selected ? "selectedActionButton" : "")} 
+            onClick={()=>this.handleClick()}
+          >
+            {isHovered && selected ? "" : name.length > 14 ? name.slice(0,11)+"...": name}
```
Comment (adrianq) [inline components/general/NetPyNEThumbnail.js]: Can we use a regular if/else?

File: components/definition/cellRules/NetPyNECellRules.js [hunk 12]
```diff
@@ -286,6 +320,29 @@ export default class NetPyNECellRules extends React.Component {
     }
   }
 
+  handleHierarchyClick = (nextPage) => {
+    const { page, selectedCellRule, value } = this.state;
```
Comment (adrianq) [inline components/definition/cellRules/NetPyNECellRules.js]: I am bit scared of this method. Can we simplify it somehow? Maybe change the swithc with a if/else and double check if we can simplify some conditions or merge two of them... Some comments will be helpful when we look at this code in a few weeks as well

**Review by adrianq — CHANGES_REQUESTED** [review]
@rodriguez-facundo  User experience is much better and the graphical interfaces looks really nice. Tests are passing as well. I have dropped a few comments regarding the code. Hopefully they are not too painful and will help to read and maintain the code in the future. Please have a look at them and let me know if you want to discuss anything. Thanks!

**Review by adrianq — APPROVED** [review]

**Pull request merged**